ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -508,6 +520,181 @@ const triggerSet: TriggerSet<Data> = {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('SE', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('SE', 'light');
+        }
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Beam',
+      type: 'StartsUsing',
+      netRegex: { id: ['82F[12]'], source: 'Anthropos' },
+      condition: (data) => data.engravementCounter === 1,
+      alertText: (data, matches, output) => {
+        if (data.me === matches.target) {
+          if (matches.id === '82F1')
+            return output.lightBeam!();
+          return output.darkBeam!();
+        }
+      },
+      outputStrings: {
+        lightBeam: {
+          en: 'light beam',
+          cn: '引导光激光',
+        },
+        darkBeam: {
+          en: 'dark beam',
+          cn: '引导暗激光',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Tower Drop Notifier',
+      type: 'GainsEffect',
+      netRegex: { effectId: ['DFB', 'DFC'] },
+      condition: (data) => data.engravementCounter === 1,
+      delaySeconds: 1, // Make sure Drop Tower Pos Tracker is done
+      alertText: (data, matches, output) => {
+        data.engravement1DarkBeamsPos = [];
+        data.engravement1LightBeamsPos = [];
+        if (data.me === matches.target) {
+          data.engravement1BeamsPosMap.forEach((value: string, key: string) => {
+            if (matches.effectId === 'DFB' && value === 'light') {
+              if (key === 'NE')
+                data.engravement1LightBeamsPos.push(output.northeast!());
+              else if (key === 'NW')
+                data.engravement1LightBeamsPos.push(output.northwest!());
+              else if (key === 'SE')
+                data.engravement1LightBeamsPos.push(output.southeast!());
+              else if (key === 'SW')
+                data.engravement1LightBeamsPos.push(output.southwest!());
+            } else if (matches.effectId === 'DFC' && value === 'dark') {
+              if (key === 'NE')
+                data.engravement1DarkBeamsPos.push(output.northeast!());
+              else if (key === 'NW')
+                data.engravement1DarkBeamsPos.push(output.northwest!());
+              else if (key === 'SE')
+                data.engravement1DarkBeamsPos.push(output.southeast!());
+              else if (key === 'SW')
+                data.engravement1DarkBeamsPos.push(output.southwest!());
+            }
+          });
+
+          // if light tower
+          if (matches.effectId === 'DFB') {
+            return output.lightTowerSide!({
+              pos1: data.engravement1LightBeamsPos[0],
+              pos2: data.engravement1LightBeamsPos[1],
+            });
+          }
+
+          return output.darkTowerSide!({
+            pos1: data.engravement1DarkBeamsPos[0],
+            pos2: data.engravement1DarkBeamsPos[1],
+          });
+        }
+      },
+      outputStrings: {
+        lightTowerSide: {
+          en: 'Drop light tower at ${pos1}/${pos2}',
```

Comment:
```suggestion
          en: 'Drop light tower ${pos1}/${pos2}',
```

Just making this shorter.